web/src/views/create-happening.js

```diff
@@ -0,0 +1,183 @@
+import React from 'react';
+import Field from 'components/form/field';
+import Label from 'components/form/label';
+import Control from 'components/form/control';
+import TextInput from 'components/form/text-input';
+import Validation from 'components/form/validation';
+
+import extend from 'lib/extend'
+import reject from 'lib/reject';
+import isEmpty from 'lib/is-empty';
+import pipe from 'lib/pipe';
+import SubmitButton from 'components/submit-button';
+import Form from 'components/form/form';
+import Button from 'components/button';
+
+import ModalCard from 'components/modal/modal-card';
+import ModalCardHead from 'components/modal/modal-card-head';
+import ModalCardContent from 'components/modal/modal-card-content';
+import ModalCardFoot from 'components/modal/modal-card-foot';
+import HappeningService from 'services/happening-service';
+
+export default function CreateHappeningView ({state, setState, onCreate, onClose}) {
+  var validateName = () => {
```

Comment:
this should be a class extending component with its own state, you should avoid passing other component's state directly as prop to child components